scripts/src/middleware.ts

```diff
@@ -8,8 +8,14 @@ import { getConfig } from "./config";
 import { generateId, pick } from "./utils";
 import { MarketplaceError } from "./errors";
 import { getDefaultLogger } from "./logging";
+import { abort as restartServer } from "./server";
 
 const START_TIME = (new Date()).toISOString();
+
+const RESTART_ERROR_COUNT = 5;  // Amount io errors to occur in time frame to trigger restart
```

Comment:
why io errors? can't it be any unknown error that causes 5xx?

Reply:
lol, spelling mistake. suppose to be "of"